### ISSUE TITLE
Import Products - Exception "Sequence contains no matching element" when removing unpublished category

### ISSUE THREAD

**Issue by noptools** [nopSolutions/nopCommerce#5528]
See post [https://www.nopcommerce.com/en/boards/topic/57290/excel-import-error/page/2#283051](url)

nopCommerce version: 4.30 (although I think the same will apply to 4.40)

Steps to reproduce the problem: 
-Create a product having a Category that is unpublished.
-Product List > Export that product to Excel
-Edit the exported Excel file.  Remove the unpublished category from the Categories column
-Product List > Import the Excel

The problem is in the code in \Libraries\Nop.Services\ExportImport\ImportManager.cs
```
//delete product categories
var deletedProductCategories = categories.Where(categoryId => !importedCategories.Contains(categoryId))
    .Select(categoryId => _categoryService.GetProductCategoriesByProductId(product.Id).First(pc => pc.CategoryId == categoryId));
```
 
The showHidden should be passed in as 'true'"
_categoryService.GetProductCategoriesByProductId(product.Id, **showHidden:true**)

I suspect 'backend' functions should see unpublished categories, so, you should also check
ExportManager.cs: ExportProductsToXml()
-  var productCategories = _categoryService.GetProductCategoriesByProductId(product.Id);
-  (the export to xlsx uses helper GetCategories() which does have 'true')

CopyProductService.cs:  CopyCategoriesMapping()

But even 'front end' calculations could be impacted - e.g. (although I don't recall if the HasCategory discount requirement allows selecting an unpublished category:
PriceCalculationService.cs:  GetAllowedDiscountsAppliedToCategories()

**Comment by skoshelev** [comment]
Closed #5528

**Comment by noptools** [comment]
I see you fixed this in ExportManager  and ImportManager.  Did you consider looking at other areas of the program - e.g. CopyProductService?
